### ISSUE TITLE
[Bug]: Stuttering on videos with lower framerate than screen refresh rate

### ISSUE THREAD

**Issue by leogh73** [nova-video-player/aos-AVP#1164]
I have an Nvidia Shield with Android TV (v11), 4K resolution at 23.97 hz. I have Nova Player lastest version. I've been having stuttering with some videos with different framerate than 23.97:

- 25 fps and over: with adaptive refresh rate enabled, they play perfectly fine. With adaptive refresh rate disabled, there's stuttering unsurprisingly.

- 23.8 and lower: with adaptive refresh rate enabled, there's stuttering. With adaptive refresh rate disabled, they play perfectly fine.

I tried 4K resolution with 59.97 hz. refresh rate. The stuttering it's gone with videos on 30 fps and lower, but I get this awful soap opera effect on some YouTube videos. I don't want to see them like that, even if that's what the creator intended.

It's there some way fix to this?? I'm not sure if this is a bug or a limitation of the OS, because I know Android 12 implements the Frame Rate Matching feature.

Thanks in advace. Kind regards.

**Comment by courville** [comment]
Thanks for reporting.
Note that Nvidia Shield is not Android 12 atm.
Can you please share which TV you are using and now the nvidia shield is connected (via AV receiver, sound bar, directly, which TV, supported refresh rates)?
When using 23.976fps and you experience stuttering, do you have a possibility to check to which refresh rate the TV is switching?
If you can share a specific video sample to reproduce this would be nice.

**Comment by leogh73** [comment]
> Can you please share which TV you are using and now the nvidia shield is connected (via AV receiver, sound bar, directly, which TV, supported refresh rates)?

My TV is LG OLED C2 EVO and the Shield is connect through an HDMI cable. I know the TV supports up to 120 hz. I set the TrueMotion feature to "Cinematic Movement", there's soap opera effect otherwise.

> When using 23.976fps and you experience stuttering, do you have a possibility to check to which refresh rate the TV is switching? 

I have no idea how to check that. If you can point me to some Android app to do it, I'll give it a shot.

> If you can share a specific video sample to reproduce this would be nice.

To be honest, I had the stuttering problem with one video so far, but I know it can happen again anytime (I download content from different sources). Anyway, I tried 2 video editors, but they crashed when I tried to extract 3 minutes of this movie. I'll keep trying on that one.
Here's some information of the file if it helps:
  Códec: HVC1 - Interno FFmpeg(hevc, Thread Frame)
  Input: HVC1(24 bits)
  Input size: 3840 × 2072(1.85:1)
  Output type: NV12(12 bits)
  Output size: 3840 × 2072(1.85:1)
  Frames/seg: 23.81
  BitRate: Unknown

**Comment by leogh73** [comment]
I tried at least 10 different versions of Nova: from 5.15.21 from 6.2.65. I noticed that with version 6.2.26 the stuttering it's almost non-existant. And my perception it's that with the earlier versions the stuttering it's a little less noticeable that with later versions. I hope it helps.

**Comment by courville** [comment]
Thanks for the feedback. One thing perhaps to test is to remove all motion compensation features enabled on the TV to see if there is no bad combination with nova rate switching.

**Comment by leogh73** [comment]
> Thanks for the feedback. One thing perhaps to test is to remove all motion compensation features enabled on the TV to see if there is no bad combination with nova rate switching.

I tried turning that off but setting the screen refresh rate at 59.97 hz., It seems a little better but I have to keep testing it. :)

**Comment by courville** [comment]
@leogh73, I would advise not to force any refresh rate on the TV and letting nova do the refresh rate setting according to content fps.

**Comment by leogh73** [comment]
> @leogh73, I would advise not to force any refresh rate on the TV and letting nova do the refresh rate setting according to content fps.

Precisely, to do that, I set the refresh rate on 59.97 hz. That's the video signal from the Nvidia Shield, not the TV itself (the TV doesn't have settings for that). The Nvidia Shield needs some setting for the screen, that includes the refresh rate. Before I use it on 23.97 hz. but if the content was at 25 or 30 fps, I guess Nova couldn't set the correct refresh rate, because the video signal was capped to 23.97. On top of that, the "Cinematic Movement" setting for the TrueMotion feature, I don't think it helped.
Anyway, I'll keep you posted.
Thank you for your time. :)
